envisage/plugins/ipython_kernel/kernelapp.py

```diff
@@ -342,6 +343,13 @@ def close_sockets(self):
             self.context.term()
             self.log.debug("Terminated zmq context")
 
+    def close_profile_dir(self):
+        """
+        Undo changes made in init_profile_dir.
+        """
+        if self.ipython_dir in sys.path:
+            sys.path.remove(self.ipython_dir)
```

Comment:
Not quite right: this needs an `os.path.abspath`.

Note: this fix isn't perfectly correct on Python 2, where we should be encoding the path with the filesystem encoding (which is [what IPython does](https://github.com/ipython/ipython/blob/033ab93c732772148953ce8d34ee4f7c0e60485a/IPython/core/application.py#L276-L277)). But given that we're due to drop Python 2 support soon, I'm honestly not sure it's worth adding that complication to the code.